clients/drcachesim/simulator/cache_stats.cpp

```diff
@@ -43,7 +43,8 @@ cache_stats_t::cache_stats_t(const std::string &miss_file, bool warmup_enabled)
 }
 
 void
-cache_stats_t::access(const memref_t &memref, bool hit, addr_t replaced_block)
+cache_stats_t::access(const memref_t &memref, bool hit,
+                      caching_device_block_t *cache_block)
```

Comment:
Maybe this should be read-only `const &`?  I would not expect the stats class to modify the block.

Reply:
It should not be const. This would allow us to extend the caching_device_block_t class to hold additional stats per block. Any class extending the stats class would then be able to access and update these per-block stats.